truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/JavaFunctionObject.java

```diff
@@ -24,26 +24,27 @@
  */
 package com.oracle.truffle.api.interop.java;
 
-import com.oracle.truffle.api.frame.VirtualFrame;
+import java.lang.reflect.Method;
+
 import com.oracle.truffle.api.interop.ForeignAccess;
-import com.oracle.truffle.api.nodes.RootNode;
-import java.util.List;
+import com.oracle.truffle.api.interop.TruffleObject;
+
+final class JavaFunctionObject implements TruffleObject {
+    final Method method;
+    final Object obj;
 
-final class JavaObjectMethodNode extends RootNode {
-    @Child ReadFieldNode readField;
-    @Child JavaFunctionNode function;
+    JavaFunctionObject(Method method, Object obj) {
+        this.method = method;
+        this.obj = obj;
+    }
 
-    JavaObjectMethodNode() {
-        super(JavaInteropLanguage.class, null, null);
-        readField = new ReadFieldNode();
-        function = new JavaFunctionNode();
+    public static boolean isInstance(TruffleObject obj) {
+        return obj instanceof JavaFunctionObject;
     }
 
     @Override
-    public Object execute(VirtualFrame frame) {
-        JavaInterop.JavaFunctionObject f = (JavaInterop.JavaFunctionObject) readField.execute(frame);
-        List<Object> args = ForeignAccess.getArguments(frame);
-        return JavaFunctionNode.execute(f, args.subList(1, args.size()).toArray());
+    public ForeignAccess getForeignAccess() {
+        return JavaFunctionObjectForeign.createAccess();
     }
 
-}
+} // end of JavaFunctionObject
```

Comment:
I prefer // end of class comments, as they help to understand what the closing brace is closing.